File: Smallest_SubArray_With_Min_SUM.py
def smallest(arr, target):
    current_wind_sum = 0
    min_wind_sum = (2**31)-1
    start_wind = 0
    for end_wind in range(len(arr)):
        current_wind_sum += arr[end_wind]
        while current_wind_sum >= target:
            min_wind_sum = min(min_wind_sum, end_wind - start_wind + 1)
            # "windEnd - windStart + 1"
            # This is To Get the size of current Window
            current_wind_sum -= arr[start_wind]
            start_wind += 1
            # currentWindSum -= arr[windStart]; 
            # This is To Shrink the Current Window from Left 
        
    return min_wind_sum

print(smallest([3,2,5,6,7,8,3,6,7,3], 8)) # 1

File: smallest_subArray_with_MIN_Sum.cpp
// Online C++ compiler to run C++ program online
#include <iostream>
using namespace std;
#include <climits>
#include <cmath>
int windStart = 0;
int currentWindSum = 0;
int minWindSum = INT_MAX;

int smallest(int arr[10],int size, int target){
    for (int windEnd = 0; windEnd < size; windEnd++){
        currentWindSum += arr[windEnd];
        while(currentWindSum >= target){
            minWindSum = min(minWindSum, windEnd - windStart + 1);
            /* windEnd - windStart + 1 
                This is To Get the size of current Window
            */
            currentWindSum -= arr[windStart]; 
              /* currentWindSum -= arr[windStart]; 
                This is To Shrink the Current Window from Left 
            */
            windStart++; // THis is to move the WindStart one step // Next
        }
    }
    //cout<<currentWindSum;
    return minWindSum;
}
int main() {
    int arr[10] = {3,2,5,6,7,8,3,6,7,3};
    cout<<smallest(arr, 10, 8);

    return 0;
}
